centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6Parameters.php

```diff
@@ -0,0 +1,157 @@
+     * @param bool $isEncrypted
+     *
+     * @throws AssertionException
+     */
+    public function __construct(
+        array $parameters,
+        private bool $isEncrypted = false
+    ){
+        /** @var _VmWareV6Parameters $parameters */
+        Assertion::range($parameters['port'], 0, 65535, 'parameters.port');
+        foreach ($parameters['vcenters'] as $index => $vcenter) {
+            // Validate min length
+            Assertion::notEmptyString($vcenter['name'], "parameters.vcenters[{$index}].name");
+            Assertion::notEmptyString($vcenter['username'], "parameters.vcenters[{$index}].username");
+            Assertion::notEmptyString($vcenter['password'], "parameters.vcenters[{$index}].password");
+            Assertion::notEmptyString($vcenter['url'], "parameters.vcenters[{$index}].url");
+
+            // Validate max length
+            Assertion::maxLength($vcenter['name'], self::MAX_LENGTH, "parameters.vcenters[{$index}].name");
+            Assertion::maxLength($vcenter['username'], self::MAX_LENGTH, "parameters.vcenters[{$index}].username");
+            Assertion::maxLength($vcenter['password'], self::MAX_LENGTH, "parameters.vcenters[{$index}].password");
+            Assertion::maxLength($vcenter['url'], self::MAX_LENGTH, "parameters.vcenters[{$index}].url");
+
+            // Validate specific format
+            Assertion::urlOrIpOrDomain($vcenter['url'], "parameters.vcenters[{$index}].url");
+        }
+        $this->parameters = $parameters;
+    }
+
+    /**
+     * @return _VmWareV6Parameters
+     */
+    public function getData(): array
+    {
+        return $this->parameters;
+    }
+
+    /**
+     * @param EncryptionInterface $encryption
+     *
+     * @return _VmWareV6Parameters
+     */
+    public function encryptCredentials(EncryptionInterface $encryption): array
+    {
+        if (true === $this->isEncrypted) {
+            return $this->parameters;
+        }
+
+        $encryption->setSecondKey(self::SECOND_KEY);
+        $parameters = $this->parameters;
+
+        foreach ($parameters['vcenters'] as $index => $vcenter) {
+            $parameters['vcenters'][$index]['username'] = str_starts_with($vcenter['username'], VaultConfiguration::VAULT_PATH_PATTERN)
+                ? $vcenter['username']
+                : $encryption->crypt($vcenter['username']);
+            $parameters['vcenters'][$index]['password'] = str_starts_with($vcenter['password'], VaultConfiguration::VAULT_PATH_PATTERN)
+                ? $vcenter['password']
+                : $encryption->crypt($vcenter['password']);
+        }
+
+        return $parameters;
+    }
+
+    /**
+     * @param EncryptionInterface $encryption
+     *
+     * @return _VmWareV6Parameters
+     */
+    public function decryptCredentials(EncryptionInterface $encryption): array
+    {
+        if (false === $this->isEncrypted) {
+            return $this->parameters;
+        }
+
+        $encryption->setSecondKey(self::SECOND_KEY);
```

Comment:
Maybe add EncryptionInterface so you could setSecondKey in constructor and avoid to setSecondkey in both encrypt/decrypt methods